DatadogSessionReplay/Sources/Writers/ResourcesWriter.swift

```diff
@@ -16,25 +16,98 @@ internal protocol ResourcesWriting {
+                    forKey: Constants.knownResourcesKey
+                )
+            }
+        }
+    }
 
     init(
-        core: DatadogCoreProtocol
+        core: DatadogCoreProtocol,
+        dataStoreResetTime: TimeInterval = TimeInterval(30).days
     ) {
-        self.core = core
+        self.scope = core.scope(for: ResourcesFeature.self)
+        self.telemetry = core.telemetry
+
+        self.scope?.dataStore.value(forKey: Constants.storeCreationKey) { result in
+            if let storeCreation = result.data()?.asTimeInterval(), Date().timeIntervalSince1970 - storeCreation < dataStoreResetTime {
+                self.scope?.dataStore.value(forKey: Constants.knownResourcesKey) { [weak self] result in
+                    switch result {
+                    case .value(let data, _):
+                        if let knownIdentifiers = data.asKnownIdentifiers() {
+                            self?.knownIdentifiers.formUnion(knownIdentifiers)
+                        }
+                    case .error(let error):
+                        self?.telemetry.error("Failed to read processed resources from data store: \(error)")
+                    case .noValue:
+                        break
+                    }
+                }
+            } else { // Reset if store was created more than 30 days ago
+                self.scope?.dataStore.setValue(
+                    Date().timeIntervalSince1970.asData(),
+                    forKey: Constants.storeCreationKey
+                )
+                self.scope?.dataStore.removeValue(forKey: Constants.knownResourcesKey)
+            }
+        }
     }
 
     // MARK: - Writing
 
     func write(resources: [EnrichedResource]) {
-        guard let scope = core?.scope(for: ResourcesFeature.self) else {
-            return
-        }
-        scope.eventWriteContext { _, recordWriter in
-            resources.forEach {
-                recordWriter.write(value: $0)
+        scope?.eventWriteContext { [weak self] _, recordWriter in
+            let unknownResources = resources.filter { self?.knownIdentifiers.contains($0.identifier) == false }
+            for resource in unknownResources {
+                recordWriter.write(value: resource)
             }
+            self?.knownIdentifiers.formUnion(Set(unknownResources.map { $0.identifier }))
+        }
+    }
+
+    enum Constants {
+        static let knownResourcesKey = "known-resources"
+        static let storeCreationKey = "store-creation"
+    }
+}
+
+extension Data {
+    func asTimeInterval() -> TimeInterval? {
+        var value: TimeInterval = 0
+        guard count >= MemoryLayout.size(ofValue: value) else {
+            return nil
+        }
+        _ = Swift.withUnsafeMutableBytes(of: &value) {
+            copyBytes(to: $0)
+        }
+        return value
+    }
+
+    func asKnownIdentifiers() -> Set<String>? {
+        return try? JSONDecoder().decode(Set<String>.self, from: self)
+    }
```

Comment:
question/ How will we know if it works? By silently failing on decoding (same for encoding), this may consistently not work for some clients, falling back to no resource ID caching and we wouldn't know that.